File: Algorithm/Sort/selectionSort.dart
void change(List<int> arr, int idx1, int idx2) {
  int temp = arr[idx1];
  arr[idx1] = arr[idx2];
  arr[idx2] = temp;
}

List<int> selectionSort(List<int> arr) {
  for (int i = 0; i < arr.length - 1; i++) {
    int minIdx = i;

    for (int j = i + 1; j < arr.length; j++) {
      if (arr[minIdx] > arr[j]) {
        minIdx = j;
      }
    }

    if (i != minIdx) {
      change(arr, i, minIdx);
    }
  }

  return arr;
}

void main() {
  List<int> arr = [5, 1, 3, 6, 7, 4, 2];
  List<int> sortedArr = selectionSort(arr);
  print(sortedArr);

  List<int> arr2 = [2, 1];
  List<int> sortedArr2 = selectionSort(arr2);
  print(sortedArr2);

  List<int> arr3 = [1];
  List<int> sortedArr3 = selectionSort(arr3);
  print(sortedArr3);
}

File: algorithm/sort/selection_sort.py
from unittest import TestCase


def selection_sort(arr):
    for i in range(len(arr)):
        min_idx = i
        for j in range(i + 1, len(arr)):
            if arr[min_idx] > arr[j]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

    return arr


class SelectionSortTest(TestCase):
    def test_selection_sort(self):
        arr = [2, 3, 4, 1, 1, 5, 5, 3, 4, 6, 8, 9, 7]
        sorted_arr = selection_sort(arr)
        self.assertEqual(sorted(arr), sorted_arr)
